### ISSUE TITLE
[1.10.2] Inconsistent traits (Bows/Arrows)

### ISSUE THREAD

**Issue by SirKayten** [SlimeKnights/TinkersConstruct#2910]
In the book "Materials and You" it says bows made from manyullyn bow limbs will get the "Cold-Blooded" trait. However, the actual bow gets Cold-Blooded as well as the "Insatiable" trait (which can be explained by looking at the materials "head" trait but that appears to be accidental). The downside is that both traits only work while hitting the foe with the bow itself but won't work if actually shooting arrows.

Is this behavior intended?
If so, it's kind of misleading. I've talked to some users and they expected the Cold-Blooded trait to affect the arrow hitting the enemy rather than having no meaningful effect at all. 

In short:
1. Bows get multiple traits through limbs even if there's just one written in the book. (See manyullyn, cobalt, iron, etc.)
2. Some traits don't work as expected.

Forge 2221
Mantle 1.1.5
TC 2.6.3

![book](https://cloud.githubusercontent.com/assets/28689950/26038698/615edd78-390e-11e7-9e7b-2b6d5ced99c3.PNG)
![bow1](https://cloud.githubusercontent.com/assets/28689950/26038697/615e6f50-390e-11e7-8f72-6b6ec10485e0.png)
![bow2](https://cloud.githubusercontent.com/assets/28689950/26038699/6160e01e-390e-11e7-8360-0c75e9f7eba8.png)

And thank you for this mod, it's one of the most beloved by many people. :)

**Comment by bonii-xx** [comment]
Hm, interesting. I'm using the same logic everywhere to ensure it's correct.. looks like something slipped up in this case.

**Comment by SirKayten** [comment]
Maybe I should've been a bit more specific: Getting multiple traits by one single material affects (almost) _every_ ranged weapon. Shortbows, longbows, crossbows, shuriken and arrows. Not sure about bolts though, they seem to be unaffected.
Tinkers code seems to add every trait from multiple material types (e.g. head, extra and projectile for shuriken). A material defining different traits per type will get more than one of its traits added to the resulting ranged weapon. Take a look at [Shuriken](https://github.com/SlimeKnights/TinkersConstruct/blob/fcd3e41f3376425ab21bfb9a953ab9594ebeb980/src/main/java/slimeknights/tconstruct/tools/ranged/item/Shuriken.java#L29) and [PartMaterialType](https://github.com/SlimeKnights/TinkersConstruct/blob/fcd3e41f3376425ab21bfb9a953ab9594ebeb980/src/main/java/slimeknights/tconstruct/library/tinkering/PartMaterialType.java#L110). Changing these should make a difference but I don't know if it breaks something else.

Added some additional screenshots below.
![shuriken](https://cloud.githubusercontent.com/assets/28689950/26068681/96a28b66-399e-11e7-80ce-db182917b8d8.png)
![arrow](https://cloud.githubusercontent.com/assets/28689950/26068685/9908b218-399e-11e7-9909-de735fbe4013.png)
![crossbow](https://cloud.githubusercontent.com/assets/28689950/26068695/a503c06c-399e-11e7-8f75-67af5b087e52.png)

**Comment by bonii-xx** [comment]
Shuriken getting both is intentional. It depends on which stats of the part are getting used. It's ok like that, but the display needs to be consistent.

**Comment by SirKayten** [comment]
I see. Just a suggestion though: At the moment, the head material type appears as a "general" material type. Pretty much every tool is forced to get its traits, even if unwanted (programming-wise) or just plain non-functional (e.g. bows with damage modifiying traits like Insatiable or Cold-Blooded). Removing this limitation by moving said traits to its specific material type counterpart would allow for more unique traits per material and tool. **Additionally, this fixes the entries in the book too!**

Just an example:
```java
sampleMaterial.addTrait(new TraitInsatiable(), MaterialTypes.HEAD); // ONLY for melee weapons/tools
sampleMaterial.addTrait(new TraitColdBlooded(), MaterialTypes.PROJECTILE); // ONLY for ranged weapons
sampleMaterial.addTrait(new TraitLightweight(), MaterialTypes.BOW); // ONLY for launchers

// Adding a second trait for the SAME material type (bow), which WILL be listed in the book too!
sampleMaterial.addTrait(new TraitDense(), MaterialTypes.BOW);
```
![samplematerial](https://cloud.githubusercontent.com/assets/28689950/26120279/75e6e9a8-3a70-11e7-97e9-c1a8f1a2a291.PNG)
Shuriken could either be kept as an exception or get their own material type.

**Comment by bonii-xx** [comment]
> just plain non-functional (e.g. bows with damage modifiying traits like Insatiable or Cold-Blooded).

You can always hit things with bows, I'm not even sure if there are any nonfunctional traits at all.

> trait per type

That's more or less already how it is, but there isn't really enough variation for bows that aren't simply stats.  So they just use the general ones.

**Comment by SirKayten** [comment]
> You can always hit things with bows, I'm not even sure if there are any nonfunctional traits at all.

Sure, but who actually **_does_** that? The point is, both of the traits have no effect if the bow is used the regular way. Transfering (some) **_effects_** (not the traits themselves) over to the projectile when shot could fix this. But that's not the main problem here.
> there isn't really enough variation for bows that aren't simply stats. 

In the future, if you actually want to remove Cold-Blooded from bows because you found something more useful, you won't be able to remove it without removing it from melee weapons too, as it's bound to the MaterialTypes.HEAD. This restriction (HEAD getting shared by multiple tool types) hinders addon developers (and probably Tinkers' itself) to add more customized traits per weapon/material type.
If changed appropriately, the current distribution of traits across MaterialTypes can be mimicked **_easily_**, so there's no reason to be feared of breaking something, **_plus_** giving addon developers more freedom to add traits and it additionally **_fixes_** the books bow category!
The only downside is, that current addons would have to undergo **_slight_** adjustments but they won't break without them. C'mon, this has to be an argument worth thinking about it!

Sorry for bothering you with my thoughts, but it really seems like a useful addition to me. :) 
However, if I couldn't convince you of my idea, I'm fine your decision. I think I was able to show all advantages and disadvantages to you to check whether it's worthy or not.

**Comment by bonii-xx** [comment]
>Sure, but who actually does that? The point is, both of the traits have no effect if the bow is used the regular way (shooting an arrow/bolt). Transfering (some) effects (not the traits themselves) over to the projectile when shot could fix this. But that's not the main problem here.

Same goes for many materials. But that's the whole point - that you CAN. Not that it's the best.

**Comment by bonii-xx** [comment]
Ok, this is highly nontrivial to fix. Meh. Technically it's correct.. but it's a bit convoluted.

**Comment by bonii-xx** [comment]
Closing since I know it'll never happen >_>
